### PR TITLE
src/ceph_mds.cc: [Cleanup] Drop unnecessary using declaration, etc

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/ceph_mds.cc [hunk 1]
```diff
@@ -101,28 +96,30 @@ int main(int argc, const char **argv)
...
+  auto cct = global_init(nullptr, args,
 			 CEPH_ENTITY_TYPE_MDS, CODE_ENVIRONMENT_DAEMON,
 			 0, "mds_data");
   ceph_heap_profiler_init();
 
   std::string val, action;
-  for (std::vector<const char*>::iterator i = args.begin(); i != args.end(); ) {
+  auto i = args.begin();
+  auto end = args.end();
+  for (; i != end; ) {
```
Comment (ukernel) [inline src/ceph_mds.cc]: why not for (auto i : args)

Comment (shinobu-x) [reply src/ceph_mds.cc]: @ukernel `auto i : args` doesn't work because `ceph_argparse_double_dash` expects iterator to be passed.

Comment (ukernel) [reply src/ceph_mds.cc]: why not 'for (auto i= args.begin();  i != args.end(); )'

Comment (batrick) [reply src/ceph_mds.cc]: Agreed with @ukernel, it has the benefit of limiting the scope of `i` and `end`.

**Comment by batrick** [comment]
ping @shinobu-x

**Comment by shinobu-x** [comment]
@batrick agreed

**Comment by shinobu-x** [comment]
test this please

File: src/ceph_mds.cc [hunk 2]
```diff
@@ -101,28 +96,30 @@ int main(int argc, const char **argv)
...
-  auto cct = global_init(NULL, args,
+  auto cct = global_init(nullptr, args,
 			 CEPH_ENTITY_TYPE_MDS, CODE_ENVIRONMENT_DAEMON,
 			 0, "mds_data");
   ceph_heap_profiler_init();
 
   std::string val, action;
-  for (std::vector<const char*>::iterator i = args.begin(); i != args.end(); ) {
+  auto i = args.begin();
+  auto end = args.end();
```
Comment (batrick) [inline src/ceph_mds.cc]: These two variables should be removed.

**Comment by batrick** [comment]
ping @shinobu-x

**Comment by batrick** [comment]
Closing this due to inactivity.

**Pull request closed without merging**